collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -1887,7 +1887,7 @@ private void handleSaveResult(FormSaveViewModel.SaveResult result) {
                         // finalized specifies that it should always be auto-sent.
                         String formId = getFormController().getFormDef().getMainInstance().getRoot().getAttributeValue("", "id");
                         String formVersion = getFormController().getFormDef().getMainInstance().getRoot().getAttributeValue("", "version");
-                        if (InstanceUploaderUtils.formShouldBeAutoSent(formsRepository, formId, formVersion, GeneralSharedPreferences.isAutoSendEnabled())) {
+                        if (InstanceUploaderUtils.shouldFormBeSent(formsRepository, formId, formVersion, GeneralSharedPreferences.isAutoSendEnabled())) {
```

Comment:
Could this not just always call `formSubmitMananer.scheduleSubmit()` at this point? I might have missed something but I think the `InstanceSubmitter` (that the `AutoSendTaskSpec` will use will do the right thing and only send if it needs to.

If that's the case we could just test `InstanceSubmitter` rather than the static method (as it will be the only place we make that decision).

Reply:
I was thinking about it as well. I didn't change it because I think it would change the default behavior a bit. Now autosend is triggered if we save a form that requires it, that's why we have that check (`InstanceUploaderUtils.shouldFormBeSent()`). If we get rid of it it will be triggered even if just one of already saved forms requires autosend.

Reply:
I know it's misleading because if you save a form that requires autosend the app will also send other already saved forms but it don't work the other way around even if all already saved forms require autosend but the one you save in this step doesn't it won't be triggered.

Reply:
I think (hope?) in most cases the distinction won't matter because the instance currently being saved will be the only one available for auto submission.

One extreme case is if there are multiple submissions that are hanging around and systematically fail AND there's a form in play that doesn't auto send. In that case there's some advantage to the existing check. That doesn't seem very likely to me so I'd be ok with the simpler implementation.

Reply:
Ok I removed that check and moved `shouldFormBeSent()` to `InstanceSubmitter`. However `shouldFormBeDeleted()` is still placed in `InstanceUploaderUtils` because it's used from different classes (InstanceUploaderTask). I noticed that the code between that `InstanceUploaderTask` and `InstanceSubmitter` is duplicated so probably it would be good to refactor it using `InstanceSubmitter` in `InstanceUploaderTask` maybe? Then both `shouldFormBeSent()` and `shouldFormBeDeleted()` would be placed in just one class (`InstanceSubmitter`). What do you think? Maybe as a separate pr if you agree?